sklearn/decomposition/online_lda.py

```diff
@@ -0,0 +1,678 @@
+
+        results = Parallel(n_jobs=n_jobs, verbose=self.verbose)(
+            delayed(_update_doc_distribution)
+            (X[idx_slice, :], self.exp_dirichlet_component_, self.doc_topic_prior_,
+             self.max_doc_update_iter, self.mean_change_tol, cal_sstats, rng)
+            for idx_slice in gen_even_slices(X.shape[0], n_jobs))
+
+        # merge result
+        doc_topics, sstats_list = zip(*results)
+        doc_topic_distr = np.vstack(doc_topics)
+
+        if cal_sstats:
+            # This step finishes computing the sufficient statistics for the M step
+            suff_stats = np.zeros(self.components_.shape)
+            for sstats in sstats_list:
+                suff_stats += sstats
+            suff_stats *= self.exp_dirichlet_component_
+        else:
+            suff_stats = None
+
+        return (doc_topic_distr, suff_stats)
+
+    def _em_step(self, X, total_samples, batch_update):
+        """
+        EM update for 1 iteration.
+        update `_component` by batch VB or online VB.
+
+        Parameters
+        ----------
+        X : array-like or sparse matrix, shape=(n_samples, n_features)
+            Document word matrix.
+
+        total_samples : integer
+            Total umber of documents. It is only used when
+            batch_update is `False`.
+
+        batch_update : boolean
+            Parameter that controls updating method.
+            `True` for batch learning, `False` for online learning.
+
+        Returns
+        -------
+        doc_topic_distr : array, shape=(n_samples, n_topics)
+            Unnormalized document topic distribution.
+        """
+
+        # E-step
+        _, suff_stats = self._e_step(X, cal_sstats=True, random_init=True)
+
+        # M-step
+        if batch_update:
+            self.components_ = self.topic_word_prior_ + suff_stats
+        else:
+            # online update
+            # In the literature, the weight is `rho`
+            weight = np.power(self.learning_offset + self.n_iter_, -self.learning_decay)
+            doc_ratio = float(total_samples) / X.shape[0]
+            self.components_ *= (1 - weight)
+            self.components_ += (weight *
+                                 (self.topic_word_prior_ + doc_ratio * suff_stats))
+
+        # update `component_` related variables
+        self.dirichlet_component_ = _log_dirichlet_expectation(self.components_)
+        self.exp_dirichlet_component_ = np.exp(self.dirichlet_component_)
+        self.n_iter_ += 1
+        return
+
+    def _check_non_neg_array(self, X, whom):
+        """
+        check & convert X to csr format, and make sure no negative value in X.
+
+        Parameters
+        ----------
+        X :  array-like or sparse matrix
+
+        """
+        X = check_array(X, accept_sparse='csr')
+        check_non_negative(X, whom)
+        return X
+
+    def partial_fit(self, X, y=None):
+        """
+        Online VB with Min-Batch update.
```

Comment:
Typo: Mini-Batch

Also please put the first line on the first of the docstring to follow PEP257:

```python
def partial_fit(self, X, y=None):
    """Online VB with Mini-Batch update.
    
    ... remaining paragraphs for the docstring
    """

This is useful to get IDE and the sphinx documentation properly pick up the first line as a short descriptions / tool tip for the method.
```